generate_contract_input.py

```diff
@@ -0,0 +1,81 @@
+import subprocess
+import sys
+import os
+import re
+
+file_path = 'rust-reference-info.txt'
+
+def parse_variables_from_file(file_path):
+    variables = {}
+    with open(file_path, 'r') as file:
+        content = file.read()
+        # Define regex patterns for matching variables
+        nova_url_pattern = r'\$NOVA_URL\s*=\s*\"(.+?)\"'
+        nova_commit_pattern = r'\$NOVA_COMMIT\s*=\s*\"(.+?)\"'
+
+        # Match variables using regular expressions
+        nova_url_match = re.search(nova_url_pattern, content)
+        nova_commit_match = re.search(nova_commit_pattern, content)
+
+        # Extract matched variables
+        if nova_url_match:
+            variables['NOVA_URL'] = nova_url_match.group(1)
+        if nova_commit_match:
+            variables['NOVA_COMMIT'] = nova_commit_match.group(1)
+
+    return variables
+
+
+def add_function_before_last_brace(file_path, function_definition):
```

Comment:
Do we use this function in new (last) version of script? If not, I would either remove it or comment it if it is needed for some future developments